File: MovieCatalogueV2/app/src/main/java/com/google/fdp/moviecataloguev2/alarm/NotificationHelper.java
package com.google.fdp.moviecataloguev2.alarm;

import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.provider.Settings;

import androidx.annotation.Nullable;
import androidx.core.app.NotificationCompat;

import com.google.fdp.moviecataloguev2.MainActivity;
import com.google.fdp.moviecataloguev2.R;

/**
 * Created by gama on 2019-08-18.
 * Addin Gama Bertaqwa
 * addingama@gmail.com
 */
public class NotificationHelper {

    public static String NOTIFICATION_CHANNEL_ID = "10001";
    public static CharSequence NOTIFICATION_CHANNEL_NAME = "NOTIFICATION_CHANNEL_NAME";

    private Context mContext;
    private NotificationManager mNotificationManager;
    private NotificationCompat.Builder mBuilder;

    public NotificationHelper(Context mContext) {
        this.mContext = mContext;
    }

    public void createNotification(@Nullable String title, @Nullable String message) {
        Intent resultIntent = new Intent(mContext, MainActivity.class);
        resultIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);

        PendingIntent resultPendingIntent = PendingIntent.getActivity(mContext,1000, resultIntent,
                PendingIntent.FLAG_UPDATE_CURRENT);

        mBuilder = new NotificationCompat.Builder(mContext, NOTIFICATION_CHANNEL_ID);
        mBuilder.setSmallIcon(R.mipmap.ic_launcher);
        mBuilder.setContentTitle(title).setContentText(message)
                .setAutoCancel(false)
                .setSound(Settings.System.DEFAULT_NOTIFICATION_URI)
                .setContentIntent(resultPendingIntent)
                .setAutoCancel(true);

        mNotificationManager = (NotificationManager) mContext.getSystemService(Context.NOTIFICATION_SERVICE);

        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
            int importance = NotificationManager.IMPORTANCE_HIGH;
            NotificationChannel notificationChannel = new NotificationChannel(NOTIFICATION_CHANNEL_ID, NOTIFICATION_CHANNEL_NAME, importance);
            notificationChannel.enableLights(true);
            notificationChannel.setLightColor(Color.RED);
            notificationChannel.enableVibration(true);
            long[] vibrationPattern = {100, 200, 300, 400, 500, 400, 300, 200, 400};
            notificationChannel.setVibrationPattern(vibrationPattern);
            mBuilder.setChannelId(NOTIFICATION_CHANNEL_ID);
            mNotificationManager.createNotificationChannel(notificationChannel);
        }

        mNotificationManager.notify((int)System.currentTimeMillis(), mBuilder.build());
    }
}

File: MovieCatalog 3/app/src/main/java/com/alfianh/moviecatalog/alarm/NotificationHelper.kt
package com.alfianh.moviecatalog.alarm

import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.Context.NOTIFICATION_SERVICE
import android.content.Intent
import android.graphics.Color
import android.provider.Settings.System.DEFAULT_NOTIFICATION_URI
import androidx.core.app.NotificationCompat
import com.alfianh.moviecatalog.MainActivity
import com.alfianh.moviecatalog.R

class NotificationHelper(private val mContext: Context) {
  private lateinit var mNotificationManager: NotificationManager
  private lateinit var mBuilder: NotificationCompat.Builder

  fun createNotification(title: String?, message: String?) {
    val resultIntent = Intent(mContext, MainActivity::class.java)
    resultIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)

    val resultPendingIntent = PendingIntent.getActivity(mContext,1000, resultIntent,
      PendingIntent.FLAG_UPDATE_CURRENT);

    mBuilder = NotificationCompat.Builder(mContext, NOTIFICATION_CHANNEL_ID)
    mBuilder.setSmallIcon(R.mipmap.ic_launcher)
    mBuilder.setContentTitle(title).setContentText(message).setAutoCancel(false).setSound(
        DEFAULT_NOTIFICATION_URI).setContentIntent(resultPendingIntent).setAutoCancel(true)

    mNotificationManager = mContext.getSystemService(NOTIFICATION_SERVICE) as NotificationManager

    if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
      val importance = NotificationManager.IMPORTANCE_HIGH
      val notificationChannel = NotificationChannel(NOTIFICATION_CHANNEL_ID,
          NOTIFICATION_CHANNEL_NAME, importance)
      notificationChannel.enableLights(true)
      notificationChannel.lightColor = Color.RED
      notificationChannel.enableVibration(true)
      notificationChannel.vibrationPattern = longArrayOf(100, 200, 300, 400, 500, 400, 300, 200,
          400)
      mBuilder.setChannelId(NOTIFICATION_CHANNEL_ID)
      mNotificationManager.createNotificationChannel(notificationChannel)
    }
    mNotificationManager.notify(System.currentTimeMillis().toInt(), mBuilder.build())
  }

  companion object {
    const val NOTIFICATION_CHANNEL_ID = "10001"
    const val NOTIFICATION_CHANNEL_NAME = "df"
  }
}
